src/Orchard.Cms.Web/Modules/Orchard.OpenId/Drivers/OpenIdSettingsDisplayDriver.cs

```diff
@@ -14,22 +18,33 @@ namespace Orchard.OpenId.Drivers
 {
     public class OpenIdSiteSettingsDisplayDriver : SiteSettingsSectionDisplayDriver<OpenIdSettings>
     {
+        private const string restartPendingCacheKey = "OpenIdSiteSettings_RestartPending";
+        private const string settingsGroupId = "open id";
+
         private readonly IOpenIdService _openIdServices;
         private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly INotifier _notifier;
+        private readonly IHtmlLocalizer<OpenIdSiteSettingsDisplayDriver> T;
+        private readonly IDynamicCache _dynamicCache;
 
         public OpenIdSiteSettingsDisplayDriver(IOpenIdService openIdServices,
-                                                IHttpContextAccessor httpContextAccessor)
+                                                IHttpContextAccessor httpContextAccessor,
+                                                INotifier notifier,
+                                                IHtmlLocalizer<OpenIdSiteSettingsDisplayDriver> stringLocalizer,
+                                                IDynamicCache dynamicCache)
         {
             _openIdServices = openIdServices;
+            _notifier = notifier;
             _httpContextAccessor = httpContextAccessor;
-        }
-        public override bool RequiresRestartAfterSaving(string groupId)
-        {
-            return (groupId == "open id");
+            _dynamicCache = dynamicCache;
+            T = stringLocalizer;
         }
 
         public override IDisplayResult Edit(OpenIdSettings settings, BuildEditorContext context)
         {
+            if (context.GroupId == settingsGroupId && _dynamicCache.GetAsync(restartPendingCacheKey).GetAwaiter().GetResult()!=null)
+                _notifier.Warning(T["This settings only will take effect after restarting the tenant."]);
```

Comment:
I didn't see what you mean with "in the body of the _memoryCache.Get() call". Get method doesn't expect a lambda like GetOrCreateAsync where I can add the notifier. 
Is it ok now?
Thank u for the review